src/audio/volume/volume.c

```diff
@@ -141,9 +135,10 @@ static enum task_state vol_work(void *data)
 				    vol <= cd->vol_min) {
 					cd->ramp_increment[i] = 0;
 					cd->volume[i] = cd->tvolume[i];
+					cd->ramp_finished = true;
+					cd->vol_ramp_active = false;
 				} else {
 					cd->volume[i] = vol;
-					again = 1;
```

Comment:
so you are removing the ramping actually? we did several cycles to ramp up to the target volume, now it will only do the step 1 and stop if it doesn't achieve to the target volume.

Reply:
Oh I get your point now, we will continue the ramping in the next .copy() iteration, this is good idea, even we can't finish the ramping target before the pipeline is trigger stopped, we still keep the target volume there so it should be fine.